### PR TITLE
chore: update psr/log

### PR DESCRIPTION
**Pull request by kidager**
## This PR
- updates the `psr/log` version requirements

### Notes
Some projects require other versions of `psr/log` than `1.*`

Example:
On a fresh Laravel 9 project, running `composer require open-feature/sdk` yields the following error:
```
Info from https://repo.packagist.org: #StandWithUkraine
./composer.json has been updated
Running composer update open-feature/sdk
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires open-feature/sdk * -> satisfiable by open-feature/sdk[0.0.2, ..., 0.1.1, 1.0.0, ..., 1.3.1].
    - open-feature/sdk[0.0.2, ..., 0.1.1, 1.0.0, ..., 1.3.1] require psr/log ^1.1 -> found psr/log[1.1.0, ..., 1.1.4] but the package is fixed to 3.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require open-feature/sdk:*" to figure out if any version is installable, or "composer require open-feature/sdk:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

### CODE REVIEW

**Comment by beeme1mr** [comment]
Relates to: https://github.com/open-feature/php-sdk/pull/41

**Comment by tcarrio** [comment]
Correct @beeme1mr . I am deprecating 7.4 support in this library shortly so this won't be a blocker soon.

See this comment for info: https://github.com/open-feature/php-sdk/pull/41#issuecomment-1361315723

@kidager thank you for the PR! I will merge this after the 7.4 deprecation and coinciding changes to the SDK are merged.

**Comment by kidager** [comment]
I missed that PR (and thus comment).
Thank you for your responsiveness 🙏

**Comment by tcarrio** [comment]
> I missed that PR (and thus comment). Thank you for your responsiveness pray

No worries! I've gotten the core SDK upgraded so that PHP 7.4 is deprecated. Next up, I'll be updated the contrib packages. For now, you should be able to rebase and make any necessary changes to support newer psr/log versions :slightly_smiling_face:

**Comment by tcarrio** [comment]
This is going to be a bit difficult I would think- since the interfaces actually do change between these versions we may have to target the latest. Perhaps dropping only version 1 may work though

**Comment by kidager** [comment]
> This is going to be a bit difficult I would think- since the interfaces actually do change between these versions we may have to target the latest. Perhaps dropping only version 1 may work though

You're right, for now I've rebased and made the changes, the pre-commit checks are passing. Will push and check if the different builds are working fine, if that's the case then bingo.
If not, I'm afraid we'll have to target only the latest version.

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "require": {
         "php": "^8",
         "myclabs/php-enum": "^1.8",
-        "psr/log": "^1.1"
+        "psr/log": "^1.1 || ^2.0 || ^3.0"
```
Comment (tcarrio) [inline composer.json:20]:
```suggestion
        "psr/log": "^2.0 || ^3.0"
```

In order to support the newer versions over the old ones.

**Comment by tcarrio** [comment]
@kidager this change is now applicable with the deprecation of PHP 7.4

I went ahead and bumped 1.1 out of the versions since it was incompatible, as we had discussed.

This unblocks utilizing OpenFeature with projects like Laravel. Thank you again for your contribution! 🎉

**Comment by tcarrio** [comment]
Well, sorry @kidager, there are more issues this caused than expected. We have the Flagd provider in open-feature/php-sdk-contrib and integration tests for the php-sdk against it in this repo- so we're seeing a bit of a chicken & egg scenario here.

If this takes much longer I'm going to branch off your changes so I'm not creating as much noise for you 🙏

**Review by tcarrio — APPROVED** [review]

**Comment by tcarrio** [comment]
And actually that was the last of the changes needed. LGTM 🚀

**Pull request merged**